src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -523,44 +457,43 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
             }
 
             // Project was changed in some way.  Let's go find the set of installed packages for it.
-            var dteProject = _workspace.TryGetDTEProject(projectId);
-            if (dteProject == null)
+            var projectGuid = _workspace.GetProjectGuid(projectId);
+            if (projectGuid == Guid.Empty)
             {
                 // Don't have a DTE project for this project ID.  not something we can query NuGet for.
                 return;
             }
 
-            var installedPackages = new MultiDictionary<string, string>();
-            var isEnabled = false;
+            var serviceContainer = _serviceProvider.GetService<IBrokeredServiceContainer, SVsBrokeredServiceContainer>();
+            var serviceBroker = serviceContainer.GetFullAccessServiceBroker();
+            var nugetService = await serviceBroker.GetProxyAsync<INuGetProjectService>(NuGetServices.NuGetProjectServiceV1).ConfigureAwait(false);
```

Comment:
that seems very reasonable.

Reply:
Yes, *always* check for a null service before dereferencing it. We have nullable-ref annotations on ISB-related APIs so if you had them turned on in this file, it would call this need out to you. :)

Reply:
added check.

Reply:
too true.  have NRTified this file.  thanks :)